src/mock/filter.js

```diff
@@ -0,0 +1,27 @@
+import {isTaskExpired, isTaskRepeating, isTaskExpiringToday} from "../utils.js";
+
+const taskToFilterMap = {
+  all: (tasks) => tasks.filter((task) => !task.isArchive).length,
+  overdue: (tasks) => tasks
+    .filter((task) => !task.isArchive)
+    .filter((task) => isTaskExpired(task.dueDate)).length,
+  today: (tasks) => tasks
+    .filter((task) => !task.isArchive)
+    .filter((task) => isTaskExpiringToday(task.dueDate)).length,
+  favorites: (tasks) => tasks
+    .filter((task) => !task.isArchive)
+    .filter((task) => task.isFavorite).length,
+  repeating: (tasks) => tasks
+    .filter((task) => !task.isArchive)
+    .filter((task) => isTaskRepeating(task.repeating)).length,
+  archive: (tasks) => tasks.filter((task) => task.isArchive).length,
+};
+
+export const generateFilter = (tasks) => {
+  return Object.entries(taskToFilterMap).map(([filterName, countTasks]) => {
+    return {
+      name: filterName,
+      count: countTasks(tasks),
+    };
+  });
```

Comment:
Будем учить новый синтаксис. В ES6 стрелочные функции умеют сходу возвращать объект. только не понятно что с брекетами (`{}` ) делать - они ж и тело функции обрамляют, и объект.
Вот так надо
```suggestion
  return Object.entries(taskToFilterMap).map(([filterName, countTasks]) => ({
      name: filterName,
      count: countTasks(tasks),
  }));
```